src/components/application_manager/src/commands/mobile/set_app_icon_request.cc

```diff
@@ -253,6 +249,10 @@ void SetAppIconRequest::on_event(const event_engine::Event& event) {
         const std::string& path =
             (*message_)[strings::msg_params][strings::sync_file_name]
                        [strings::value].asString();
+        if (is_icons_saving_enabled_) {
```

Comment:
@tkireva  is_icons_saving_enabled_ is true if it is possible to read and write in app_icons_folder. Only if result is true we can save icon in folder. It doesn't matter which info has response - according requirement "SDL must copy <syncFileName> from "AppStorageFolder" to "AppIconsFolder" ONLY in case of successfull response from HMI"